pkg/network/protocols/core/protocol.go

```diff
@@ -73,16 +75,16 @@ func (p *Protocol) SendSlotCommitment(cm *model.Commitment, to ...peer.ID) {
 }
 
 func (p *Protocol) SendAttestations(cm *model.Commitment, attestations []*iotago.Attestation, merkleProof *merklehasher.Proof[iotago.Identifier], to ...peer.ID) {
-	var iotagoAPI iotago.API
-	if len(attestations) > 0 {
-		// TODO: there are multiple attestations potentially spanning multiple epochs/versions, we need to use the correct API for each one
-		iotagoAPI = lo.PanicOnErr(p.apiProvider.APIForVersion(attestations[0].ProtocolVersion))
-	} else {
-		iotagoAPI = p.apiProvider.APIForSlot(cm.Index()) // we need an api to serialize empty slices as well
+	encodedAttestations := marshalutil.New()
+	encodedAttestations.WriteUint32(uint32(len(attestations)))
+	for _, att := range attestations {
+		iotagoAPI := lo.PanicOnErr(p.apiProvider.APIForVersion(att.ProtocolVersion))
+		encodedAttestations.WriteBytes(lo.PanicOnErr(iotagoAPI.Encode(att)))
```

Comment:
Instead of marshalutil we could use `stream.WriteCollection`

Reply:
a bytes buffer does not implement WriteSeeker, so could not use it here

Reply:
Ok then let's implement something like this https://stackoverflow.com/a/73679110 in the context of #284 so that we're able to use a bytes buffer with the stream package and replace this later